[Model Averaging] Update model averager API to avoid the redundant `params` arg needed by post-localSGD optimizer

wayi1 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #62132
* #62131

as title

Proposal: https://github.com/pytorch/pytorch/issues/59699

Differential Revision: [D29887751](https://our.internmc.facebook.com/intern/diff/D29887751/)

Commit:
[Model Averaging] Update model averager API to avoid the redundant `params` arg needed by post-localSGD optimizer

as title

Proposal: https://github.com/pytorch/pytorch/issues/59699

Differential Revision: [D29887751](https://our.internmc.facebook.com/intern/diff/D29887751/)

[ghstack-poisoned]

rohan-varma commented on torch/testing/_internal/distributed/distributed_test.py:
```diff
@@ -4376,21 +4376,21 @@ def test_post_localSGD_optimizer_parity(self, grad_is_view=False):
             loss_fn = nn.MSELoss()
 
-            for idx in range(20):
+            for step in range(20):
```
nit: `for _ in range(20)` since it is not used?

SciPioneer replied:
Sure. Previously used step and then forgot to remove it here. Hopefully this can be captured by linter in the future.

Commit:
Update on "[Model Averaging] Update model averager API to avoid the redundant `params` arg needed by post-localSGD optimizer"


as title

Proposal: https://github.com/pytorch/pytorch/issues/59699

Differential Revision: [D29887751](https://our.internmc.facebook.com/intern/diff/D29887751/)

[ghstack-poisoned]

Commit:
Update on "[Model Averaging] Update model averager API to avoid the redundant `params` arg needed by post-localSGD optimizer"


as title

Proposal: https://github.com/pytorch/pytorch/issues/59699

Differential Revision: [D29887751](https://our.internmc.facebook.com/intern/diff/D29887751/)

[ghstack-poisoned]

Commit:
Update on "[Model Averaging] Update model averager API to avoid the redundant `params` arg needed by post-localSGD optimizer"


as title

Proposal: https://github.com/pytorch/pytorch/issues/59699

Differential Revision: [D29887751](https://our.internmc.facebook.com/intern/diff/D29887751/)

[ghstack-poisoned]

Commit:
Update on "[Model Averaging] Update model averager API to avoid the redundant `params` arg needed by post-localSGD optimizer"


as title

Proposal: https://github.com/pytorch/pytorch/issues/59699

Differential Revision: [D29887751](https://our.internmc.facebook.com/intern/diff/D29887751/)

[ghstack-poisoned]
